source/write/gridfs.txt

```diff
@@ -0,0 +1,309 @@
+   :dedent:
+   :start-after: start create custom bucket
+   :end-before: end create custom bucket
+
+.. _gridfs-upload-files:
+
+Upload Files
+------------
+
+Use the ``MongoDB\GridFS\Bucket::openUploadStream()`` method to
+create an upload stream for a given file name. The ``openUploadStream()``
+method allows you to specify configuration information in an options
+array, which you can pass as a parameter to the ``openUploadStream()`` method.
+
+This example uses an upload stream to perform the following
+actions:
+
+- Opens a writable stream for a new GridFS file named ``'my_file'``
+- Sets the ``chunkSizeBytes`` and ``metadata`` options in an array parameter
+  to the ``openUploadStream()`` method
+- Calls the ``fwrite()`` method to write data to ``'my_file'``, which the stream points to
+- Calls the ``fclose()`` method to close the stream pointing to ``'my_file'`` 
+
+.. literalinclude:: /includes/write/gridfs.php
+   :language: php
+   :dedent:
+   :start-after: start upload files
+   :end-before: end upload files
+
+.. _gridfs-retrieve-file-info:
+
+Retrieve File Information
+-------------------------
+
+In this section, you can learn how to retrieve file metadata stored in the
+``files`` collection of the GridFS bucket. The metadata contains information
+about the file it refers to, including:
+
+- The ``_id`` of the file
+- The name of the file
+- The length/size of the file
+- The upload date and time
+- A ``metadata`` document in which you can store any other information
+
+To retrieve files from a GridFS bucket, call the ``MongoDB\GridFS\Bucket::find()``
+method on the ``MongoDB\GridFS\Bucket`` instance. The method returns a ``MongoDB\library\Cursor``
+instance from which you can access the results. To learn more about ``Cursor`` objects in
+the {+php-library+}, see the :ref:`<php-cursors>` guide.
+
+The following code example shows you how to retrieve and print file metadata
+from files in a GridFS bucket. It uses a ``foreach`` loop to iterate through
+the returned cursor and display the contents of the file uploaded in the
+:ref:`gridfs-upload-files` example:
+
+.. io-code-block::
+   :copyable:
+
+   .. input:: /includes/write/gridfs.php
+      :start-after: start retrieve file info
+      :end-before: end retrieve file info
+      :language: php
+      :dedent:
+
+   .. output::
+      :visible: false
+
+      {"_id":{"$oid":"..."},"chunkSize":1048576,"filename":"my_file",
+      "length":13,"uploadDate":{"$date":{"$numberLong":"..."}},"metadata":
+      {"contentType":"text\/plain"},"md5":"6b24249b03ea3dd176c5a04f037a658c"}
+
+The ``find()`` method accepts various query specifications. You can use its
+``$options`` parameter to specify the sort order, maximum number of documents to return,
+and the number of documents to skip before returning. To view a list of available
+options, see the `API documentation <{+api+}/method/MongoDBGridFSBucket-find/#parameters>`__.
+
+.. _gridfs-download-files:
+
+Download Files
```

Comment:
Consider `downloadToStream()`, from the second code example in [Downloading Files with Readable Streams](https://www.mongodb.com/docs/php-library/current/tutorial/gridfs/#downloading-files-with-readable-streams).

If we think the use case of downloading from GridFS to a file is common enough, that's the most convenient and efficient way to do so.